--- charts-original/templates/_k10_template.tpl
+++ charts/templates/_k10_template.tpl
@@ -75,11 +75,7 @@
       securityContext:
 {{ toYaml .Values.services.securityContext | indent 8 }}
       serviceAccountName: {{ template "serviceAccountName" . }}
-{{- if or .Values.secrets.dockerConfig .Values.secrets.dockerConfigPath }}
-{{- include "check.validateImagePullSecrets" .}}
-      imagePullSecrets:
-        - name: {{ .Values.global.imagePullSecret }}
-{{- end }}
+      {{- include "k10.imagePullSecrets" . | indent 6 }}
 {{- /* initContainers: */}}
 {{- (dict "main" . "k10_pod" $service | include "k10-init-container-header") }}
 {{- (dict "main" . "k10_pod" $service | include "k10-init-container") }}
